opentelemetry-sdk/src/attributes/set.rs

```diff
@@ -135,7 +141,12 @@ impl From<&Resource> for AttributeSet {
             .collect::<Vec<_>>();
         vec.sort_unstable();
 
-        AttributeSet(vec)
+        let mut hasher = DefaultHasher::new();
+        for item in &vec {
+            item.hash(&mut hasher);
+        }
+
+        AttributeSet(vec, hasher.finish())
```

Comment:
This code is duplicated above.  Suggest a private `new` method accepting a `Vec` and put the hash computation in there, once.  (Also suggest seeing if the four-line for-loop could make use of `std::iter::fold` instead... e.g. `vec.iter().fold(&mut hasher, |&mut hasher, item| item.hash(hasher))` or something.

Reply:
Done